docs/cookbook/ion-schema-rust-getting-started.md

```diff
@@ -0,0 +1,342 @@
+* [How to load a `Schema`?](#how-to-load-a-schema)
+    * [Loading a schema](#loading-a-schema)
+* [How to create a schema programmatically(`IslSchema`)?](#how-to-create-a-schema-programmaticallyislschema)
+    * [How to create an `IslType`?](#how-to-definecreate-an-isltype)
+    * [How to create `IslSchema`?](#how-to-create-islschema)
+* [How to validate an Ion value using a `Schema`?](#how-to-validate-an-ion-value-using-a-schema)
+
+## How to use this cookbook?
+This cookbook explains how to create all the necessary structures required to perform Ion schema validation. It gives sample code with each section explaining the usage of that operation or structure.
+In general, the process of Ion schema validation contains two steps. 
+1. The first step is to load a correct Ion schema. In this step, we use a `SchemaSystem` which verifies the syntax of given schema file for its correctness.
+2. The second step is to use this generated schema from #1 and validate Ion values using the type definitions defined within the schema.
+
+This cookbook starts from examples for #1(including examples for all the required structures to be created to complete #1) and at the end adds examples for #2(How to validate Ion value using `ion-schema-rust`)
+
+## Terms
+
+* **Schema Id**: A unique identifier provided to each schema.
+* **Authority:** Authority is responsible for resolving particular class of schema identifiers. One example of authority is file system authority which resolves schema ids to files relative to a base path.
+* **Schema System:**  Provides methods for instantiating instances of Schema. Requests each of the provided Authorities, in order to resolve the requested schema id until one successfully resolves it. 
+* **Schema:** Schema is a collection of 0 or more Types. It contains an optional header, type definitions/ types and an optional footer. The schema header can optionally contain imports for that schema.
+* **Import:** An Import allows types from other schemas to be used within a schema definition. These imports are usually defined in the schema header.
+
+## How to create a `SchemaSystem`?
+
+`ion-schema-rust` requires you to create an `SchemaSystem` in order to load a schema from given file location. 
+Hence, Creating a `SchemaSystem` is the first step to loading a schema file for validation. 
+
+### Creating `DocumentAuthority`:
+
+In general, users will create a `DocumentAuthority` and then use it to build the `SchemaSystem`. Then this `SchemaSystem` is used instantiating instances of `Schema`.
+There are two types of `DocumentAuthority` available:
+
+* `FileSystemDocumentAuthority` 
+    * This authority allows to specify the base path to where all the schema files are saved (e.g. `/home/USER/schemas/`)
+    * Adding this authority your `SchemaSystem` would allow to resolve all schema ids (relative path to schema file, e.g. `my_schema.isl`) that are within this authority’s base path(e.g. `/home/USER/schemas/`)
```

Comment:
```suggestion
    * Adding this authority your `SchemaSystem` would allow to resolve all schema ids (relative path to schema file, e.g. `my_schema.isl`) that are within this authority’s base path (e.g. `/home/USER/schemas/`)
```